// sms-node/lib/sendSMS.js
'use strict';

var _btoa = require('btoa');

var _btoa2 = _interopRequireDefault(_btoa);

var _requestPromise = require('request-promise');

var _requestPromise2 = _interopRequireDefault(_requestPromise);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

class SMS {
    constructor(settings) {
        this.settings = settings;
    }

    sms(body) {
        let result = [];
        var _settings = this.settings;
        const user = _settings.user,
              password = _settings.password,
              url = _settings.url;

        const auth = (0, _btoa2.default)(`${user}:${password}`);
        const authorization = `Basic ${auth}`;
        const headers = {
            'Content-Type': 'application/json',
            Accept: 'application/json',
            authorization
        };
        const defaultUrl = 'https://gateway.plusmms.net/rest/message';
        result = _requestPromise2.default.post({
            url: url || defaultUrl,
            headers,
            body: JSON.stringify(body)
        });
        return result;
    }
}

module.exports = SMS;
